src/testsuite/arith128_test_char.c

```diff
@@ -296,6 +296,308 @@ test_popcntb (void)
   return (rc);
 }
 
+//#define __DEBUG_PRINT__ 1
+int
+test_mrgeob (void)
+{
+  vui8_t i, j;
+  vui8_t k, e;
+  int rc = 0;
+
+  printf ("\ntest_mrgeob Vector Merge Even/Odd Bytes\n");
+
+  i = (vui8_t) { 10, 1, 20, 2, 30, 3, 40, 4,
+		50, 5, 60, 6, 70, 7, 80, 8 };
+  j = (vui8_t) { 90, 9, 0xa0, 10, 0xb0, 11, 0xc0, 12,
+		 0xd0, 13, 0xe0, 14, 0xf0, 15, 0xff, 0 };
+  e = (vui8_t) { 10, 90, 20, 160, 30, 0xb0, 40, 0xc0,
+		 50, 0xd0, 60, 0xe0, 70, 0xf0, 80, 0xff };
```

Comment:
Because it was easier to the math in my head. But if you insist, I'll change it.

[global] would not work in general and would generate a lot of code.

Reply:
Re: global, why?  they're all compile-time constants.

Reply:
I have tried and the compiler does not handle this well. I suspect as the PowerISA does not have vector indexed indirect. The compiler does not have a pattern for this. So it stores the vector then loads one byte at a time using the index then stores that byte. Finally it loads the vector.

Another lesson. Don't assume write, some code and see what the compiler will actually do!

Reply:
Lesson #2: if the compiler doesn't do what it should, get the compiler fixed.  I'll poke at the team.